ddl/ddl_api.go

```diff
@@ -1455,7 +1455,22 @@ func modifiable(origin *types.FieldType, to *types.FieldType) error {
 			return nil
 		}
 	case mysql.TypeEnum:
-		return errUnsupportedModifyColumn.GenWithStackByArgs("modify enum column is not supported")
+		if origin.Tp == to.Tp {
+			if len(to.Elems) < len(origin.Elems) {
+				msg := fmt.Sprintf("the number of enum column's elements is less than the original: %d", len(origin.Elems))
+				return errUnsupportedModifyColumn.GenWithStackByArgs(msg)
+			}
+			for index, originElem := range origin.Elems {
+				toElem := to.Elems[index]
+				if originElem != toElem {
```

Comment:
For enum `enum('N','Y')`, `alter table t change c2 c2 enum('N','Y','A','B') DEFAULT 'A';` is okay, but `alter table t change c2 c2 enum('A','B','N','Y') DEFAULT 'A'` does not work.

Reply:
What do you mean? The enum order shouldn't be changed. The SQL "alter table t change c2 c2 enum('A','B','N','Y') DEFAULT 'A' " is also  not working at MySQL.

Reply:
I use MySQL 5.6 to execute these case.
in MySQL:
   
    mysql> create table t (c1 varchar(64),c2 enum('N','Y') not null default 'N',c3 
    timestamp on update current_timestamp,c4 int primary key,unique key idx2 (c2, 
    c3));
    Query OK, 0 rows affected (0.03 sec)
    mysql> insert into t values('a', 'N', '2017-07-01', 8);
    Query OK, 1 row affected (0.00 sec)
    mysql> alter table t change c2 c2 enum('A','B','N','Y') DEFAULT 'A';
    Query OK, 1 row affected (0.05 sec)
    Records: 1  Duplicates: 0  Warnings: 0
in TiDB:  

    mysql> insert into t values('a', 'N', '2017-07-01', 8);
    Query OK, 1 row affected (0.00 sec)
    mysql> alter table t change c2 c2 enum('A','B','N','Y') DEFAULT 'A';
    ERROR 1105 (HY000): unsupported modify column cannot modify enum column 
    value N to A
    mysql> alter table t change c2 c2 enum('N','Y','A','B') DEFAULT 'A';
    Query OK, 0 rows affected (0.01 sec)